bypass.lua

```diff
@@ -6,19 +6,25 @@ function courseplay:isTheWayToTargetFree(self,lx,lz)
 	end;
 	local distance = 20
 	local heigth = 0.5
+  -- a world point 4 m in front of us, 0.5 m higher
 	local tx, ty, tz = localToWorld(self.cp.DirectionNode,0,heigth,4)
+  -- world direction 
 	local nx, _, nz = localDirectionToWorld(self.cp.DirectionNode, lx, 0, lz)
+  -- terrain height at 20 m further ahead of the 4 m point (not sure why can't we directly localToWorld to it, why nx,nz?)
 	local terrainHeight = getTerrainHeightAtWorldPos(g_currentMission.terrainRootNode, tx+(nx*distance), 0, tz+(nz*distance))
 	local _, ly,_ = courseplay:getDriveDirection(self.cp.DirectionNode, tx+(nx*distance), terrainHeight, tz+(nz*distance))
+  -- world normal vector towards a point 20 m ahead, considering terrain height.
```

Comment:
And this is the problem for waypoint following in mode2. This is always looking 20 meters directly ahead but on tight fields surrounded by obstacles our course generated by pathfinding may be much closer to the obstacles and may have tight corners. This here will initiate a collision avoidance even if there's no danger as the course will turn anyway.

One way to solve this is to look for obstacles between the current position and the next waypoint (instead of dead ahead)